integration/libs/telemetry/telemetry_test.go

```diff
@@ -0,0 +1,74 @@
+package telemetry_test
+
+import (
+	"context"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/databricks/cli/integration/internal/acc"
+	"github.com/databricks/cli/libs/telemetry"
+	"github.com/databricks/databricks-sdk-go/client"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// Wrapper to capture the response from the API client since that's not directly
+// accessible from the logger.
+type apiClientWrapper struct {
+	response  *telemetry.ResponseBody
+	apiClient *client.DatabricksClient
+}
+
+func (wrapper *apiClientWrapper) Do(ctx context.Context, method, path string,
+	headers map[string]string, request, response any,
+	visitors ...func(*http.Request) error,
+) error {
+	err := wrapper.apiClient.Do(ctx, method, path, headers, request, response, visitors...)
+	wrapper.response = response.(*telemetry.ResponseBody)
+	return err
+}
+
+func TestTelemetryLogger(t *testing.T) {
+	ctx, w := acc.WorkspaceTest(t)
+	ctx = telemetry.NewContext(ctx)
+
+	// Extend the maximum wait time for the telemetry flush just for this test.
+	telemetry.MaxAdditionalWaitTime = 1 * time.Hour
+	t.Cleanup(func() {
+		telemetry.MaxAdditionalWaitTime = 2 * time.Second
+	})
+
+	// Log some events.
+	err := telemetry.Log(ctx, telemetry.FrontendLogEntry{
+		DatabricksCliLog: telemetry.DatabricksCliLog{
+			CliTestEvent: telemetry.CliTestEvent{
+				Name: telemetry.DummyCliEnumValue1,
+			},
+		},
+	})
```

Comment:
Can we modify Log to accept Name directly? It seems like that is the only field there?

Then this and other cases can be written as

`telemetry.Log(ctx, telemetry.DummyCliEnumValue1)`

Reply:
There'll be other CLI events, like: https://github.com/databricks/cli/pull/2037